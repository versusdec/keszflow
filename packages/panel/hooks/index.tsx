export * from './useInvoices'
export * from './useInvoice'
export * from './useUser'
