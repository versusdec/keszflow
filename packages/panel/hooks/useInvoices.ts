import { useQuery } from 'react-query'

export interface invoiceData {
  id: number
  date: string
  name: string
  total: number
  status: string
}

export const useInvoices = () => {
  const { data, isFetching, isError } = useQuery<invoiceData[]>(
    ['invoices/list.json'],
    {
      enabled: true,
    }
  )

  return {
    data,
    isFetching,
    isError,
  }
}
