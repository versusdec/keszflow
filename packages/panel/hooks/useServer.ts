import { useQuery } from 'react-query'

export const useServer = () => {
  const { data } = useQuery(['invoice.pdf'], {
    enabled: true,
  })
  return {
    data,
  }
}
