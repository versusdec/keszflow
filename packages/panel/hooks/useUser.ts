import { useQuery } from 'react-query'

export enum UserRoles {
  Admin = 'admin',
  Accountant = 'accountant',
  Public = 'public',
}

export interface IUser {
  id: number
  name: string
  lastname: string
  patronymic: string
  email: string
  gender: 'male' | 'female' | 'other'
  language: string
  role: UserRoles
  status: string
  avatar: string
}

export const useUser = () => {
  const { data, isFetching, isError } = useQuery<IUser>(['users/me.json'], {
    enabled: true,
    refetchOnWindowFocus: false,
  })

  return { user: data, isFetching, isError }
}
