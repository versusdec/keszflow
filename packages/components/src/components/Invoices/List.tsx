import {
  Paper,
  Table,
  TableContainer,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
} from '@mui/material'
import { ActionButton, ActionButtonIcon } from '@keszflow/components'
import { invoiceData } from '@keszflow/panel/hooks'
import { format } from 'date-fns'

export interface IList {
  invoices: invoiceData[] | undefined
  openInvoiceHandler: (id: number, type: string) => void
}

export const InvoiceListItem = ({
  data,
  open,
}: {
  data: invoiceData
  open: (id: number, type: string) => void
}) => {
  return (
    <>
      <TableRow data-testid={`list_item_${data.id}`}>
        <TableCell>{data.id}</TableCell>
        <TableCell align="left">{data.name}</TableCell>
        <TableCell align="left">
          {format(new Date(data.date), 'dd.MM.yyyy')}
        </TableCell>
        <TableCell align="left">{data.total}</TableCell>
        <TableCell align="left">{data.status}</TableCell>
        <TableCell align="right">
          <ActionButton
            data-testid={`open_${data.id}`}
            onClick={() => {
              open(data.id, data.type)
            }}
            icon={ActionButtonIcon.Launch}
            tooltip={'Open'}
          />
          <ActionButton
            data-testid={`download_${data.id}`}
            onClick={() => {}}
            icon={ActionButtonIcon.Download}
            tooltip={'Download'}
          />
          <ActionButton
            data-testid={`delete_${data.id}`}
            onClick={() => {}}
            icon={ActionButtonIcon.Delete}
            tooltip={'Delete'}
          />
        </TableCell>
      </TableRow>
    </>
  )
}

export const InvoiceList = ({ invoices, openInvoiceHandler }: IList) => {
  return (
    <>
      <TableContainer component={Paper} data-testid={'list_container'}>
        <Table sx={{}}>
          <TableHead>
            <TableRow>
              <TableCell>ID</TableCell>
              <TableCell>Name</TableCell>
              <TableCell>Date</TableCell>
              <TableCell>Total</TableCell>
              <TableCell>Status</TableCell>
              <TableCell />
            </TableRow>
          </TableHead>
          <TableBody>
            {invoices &&
              invoices.map((invoice) => (
                <InvoiceListItem
                  key={invoice.id}
                  data={invoice}
                  open={openInvoiceHandler}
                />
              ))}
          </TableBody>
        </Table>
      </TableContainer>
    </>
  )
}
