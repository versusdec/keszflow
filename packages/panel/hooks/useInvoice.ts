import { useQuery } from 'react-query'
import { Invoice } from '@keszflow/components/src/components/Invoices/Create'

export const useInvoice = (id: number | undefined) => {
  const { data, isFetching, isError } = useQuery<Invoice>(
    ['invoices/get2.json', id],
    {
      enabled: !!id,
      refetchOnWindowFocus: false,
    }
  )

  return { data, isFetching, isError }
}
