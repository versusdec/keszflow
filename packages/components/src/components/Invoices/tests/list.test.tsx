import React from 'react'
import { IList, InvoiceList } from '../List'

import { fireEvent, render, screen } from '@testing-library/react'
import { mount, shallow } from 'enzyme'
import { invoiceData } from '@keszflow/panel/hooks/useInvoices'

describe('Invoice List', () => {
  /* const defaultQueryFn = async ({
                                  queryKey,
                                }: {
    queryKey: readonly unknown[] | [string]
  }) => {
    const res = await fetch(`https://api.dev.keszflow.business/${queryKey[0]}`)
    return res.json()
  }

  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        queryFn: defaultQueryFn,
      },
    },
  })
  const wrapper = ({children}: {children: PropsWithChildren<any>}) => (
    <QueryClientProvider client={queryClient}>
      {children}
    </QueryClientProvider>
  );

  const { result, waitFor } = renderHook(() => useInvoices(), { wrapper });

  await waitFor(() => result.current.isSuccess);

  expect(result.current.data).toEqual(useInvoicesResponse); */

  // jest.mock('useInvoices')

  // let response;
  /*  beforeEach(()=>{
      response = {
        data: useInvoicesData,
        isFetching: false,
        isError: false
      }
    })

    test('get invoice data', ()=>{
      // const mockUseQuery = (useInvoices as jest.MockedFunction<typeof useInvoices>).mockReturnValue(useInvoicesData)
      const ads = useInvoices()
      ad
      // expect(data).toBe(useInvoicesData)
    }) */

  let useInvoicesData: invoiceData[]
  let handler: (id: number, type: string) => void
  const getComponent = ({ invoices, openInvoiceHandler }: IList) => (
    <InvoiceList invoices={invoices} openInvoiceHandler={openInvoiceHandler} />
  )

  beforeEach(() => {
    useInvoicesData = [
      {
        id: 1,
        date: '2022-05-14T21:00:00.000Z',
        name: 'My Fancy invoice #1',
        total: 1500,
        status: 'active',
        type: 'created',
      },
      {
        id: 2,
        date: '2022-06-09T21:00:00.000Z',
        name: 'Another one',
        total: 24158,
        status: 'active',
        type: 'created',
      },
      {
        id: 3,
        date: '2022-06-09T21:00:00.000Z',
        name: 'And another',
        total: 4561,
        status: 'active',
        type: 'created',
      },
      {
        id: 4,
        date: '2022-07-09T21:00:00.000Z',
        name: 'Geez thats alot',
        total: 48945,
        status: 'active',
        type: 'created',
      },
      {
        id: 5,
        date: '2022-07-09T21:00:00.000Z',
        name: 'Somebody stop me!',
        total: 1245,
        status: 'active',
        type: 'created',
      },
      {
        id: 6,
        date: '2022-08-09T21:00:00.000Z',
        name: 'Uploaded Invoice #1',
        total: 15488,
        status: 'active',
        type: 'uploaded',
      },
    ]
    handler = jest.fn()
  })

  it('Should render correctly with Shallow', () => {
    const component = shallow(
      getComponent({ invoices: [], openInvoiceHandler: handler })
    )
    expect(component).toMatchSnapshot()
  })
  it('Should render correctly with mount', () => {
    const component = mount(
      getComponent({ invoices: [], openInvoiceHandler: handler })
    )
    expect(component).toMatchSnapshot()
  })

  test('open invoice modal btn click', () => {
    render(
      getComponent({ invoices: useInvoicesData, openInvoiceHandler: handler })
    )
    useInvoicesData.forEach((item, i) => {
      const listItem = screen.getByTestId(`list_item_${item.id}`)
      const itemOpenIcon = screen.getByTestId(`open_${item.id}`)
      expect(listItem).toBeInTheDocument()
      expect(itemOpenIcon).toBeInTheDocument()
      fireEvent.click(itemOpenIcon)
      expect(handler).toHaveBeenCalledTimes(i + 1)
      expect(handler).toHaveBeenCalledWith(item.id, item.type)
    })
  })
})
