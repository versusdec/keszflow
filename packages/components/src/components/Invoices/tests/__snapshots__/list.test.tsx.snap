// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice List Should render correctly with Shallow 1`] = `ShallowWrapper {}`;

exports[`Invoice List Should render correctly with mount 1`] = `ReactWrapper {}`;
